### PR TITLE
deidentify: update console output, revise saving MRN ID map

### PR DESCRIPTION
**Pull request by erikr**
## Issues closed
No issue

## Changelog
- [x] Only update MRN map and print updated new starting ID if there are new IDs
- [x] Displays progress through CSV files via `tqdm`

## Checklist
- [x] The following are complete:
    - Added or revised tests if appropriate.
    - If new packages are installed, `requirements.txt` is updated.
    - If documentation should be updated, a new issue was made in `ml4c3-wiki`.
    - The PR name describes its content and does not contain "Ref #".
    - The PR does not add PHI.

### CODE REVIEW

File: scripts/deidentify.py [hunk 1]
```diff
@@ -163,17 +164,20 @@ def _remap_mrns(args):
...
-    print(f"Last ID used in remapping MRNs was {df['new_id'].max()}")
+    if len(new_mrns) > 0:
+        new_ids = list(range(starting_id, len(new_mrns) + starting_id))
+        np.random.shuffle(new_ids)
+        mrn_map.update(dict(zip(new_mrns, new_ids)))
+        print(f"New MRNs remapped starting at ID {starting_id}")
+    print(f"Last ID used in remapping MRNs was {max(mrn_map.values())}")
+
+    # If no mrn_map file exists, save it
+    if not os.path.isfile(args.mrn_map):
```
Comment (StevenSong) [inline scripts/deidentify.py:175]: what about the case where the MRN map needs to be updated? it doesnt look like its getting saved in that case... the old code didn't do these checks as a naive way to simplify code i.e. always update the map - seems like these checks need to be more complex

Comment (erikr) [reply scripts/deidentify.py]: Good catch, thanks.

Now, a new map is saved every run with the file name `mgb-$TIMESTAMP.csv`, where `$TIMESTAMP` is `yyyy-mm-dd-hh:mm:ss`. This is crude but safe. Users must explicitly delete outdated maps.

**Comment by erikr** [comment]
@StevenSong I made a few changes. Now running on all of our ECGs (https://github.com/aguirre-lab/ml4c3/issues/1241) and backing up to `MAD3` since `mithril` does not have enough SSD space and the HDD is not mounted.

File: scripts/deidentify.py [hunk 2]
```diff
@@ -163,17 +165,22 @@ def _remap_mrns(args):
...
+        print(f"New MRNs remapped starting at ID {starting_id}")
+    print(f"Last ID used in remapping MRNs was {max(mrn_map.values())}")
+
+    today = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+    mrn_map_filename = args.mrn_map.replace(".csv", "")
+    mrn_map_path_new = f"{mrn_map_filename}-{today}.csv"
 
     df = pd.DataFrame.from_dict(mrn_map, orient="index", columns=["new_id"])
     df.index.name = "mrn"
-    os.makedirs(os.path.dirname(args.mrn_map), exist_ok=True)
```
Comment (StevenSong) [inline scripts/deidentify.py:173]: might need to restore this line

File: scripts/deidentify.py [hunk 3]
```diff
@@ -163,17 +165,22 @@ def _remap_mrns(args):
...
-    df.sort_values("new_id").to_csv(args.mrn_map)
-    print(f"MRN map saved to {args.mrn_map}")
-    print(f"Last ID used in remapping MRNs was {df['new_id'].max()}")
+    df.sort_values("new_id").to_csv(mrn_map_path_new, index=False)
+    print(f"MRN map saved to {mrn_map_path_new}")
```
Comment (StevenSong) [inline scripts/deidentify.py:183]: another idea is that the map really only needs to be saved anew if there are new MRNs that were remapped, which is checked in the condition above

**Review by StevenSong — APPROVED** [review]

**Comment by StevenSong** [comment]
@erikr I'm going to merge this just so @epalto can start the purge, my review comments can be addressed later

**Pull request merged**